src/Rings/mpolyquo-localizations.jl

```diff
@@ -1687,6 +1706,32 @@ end
 ## MPolyAnyIdeal they are handling
 ############################################################################# 
 
+@doc raw"""
+     primary_decomposition(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdeal, <:MPolyLocalizedIdeal})
+
+Return primary decomposition of ``I``
+"""
+function primary_decomposition(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdeal, <:MPolyLocalizedIdeal})
+  Q = base_ring(I)
+  R = base_ring(Q)
+  decomp = primary_decomposition(saturated_ideal(I))
+  result = [(ideal(Q, Q.(gens(a))), ideal(Q, Q.(gens(b)))) for (a, b) in decomp]
```

Comment:
Some of these ideals could be `(1)` in the localized case or agree. 
Does this match the docstring/definition for `primary_decomposition`? 
And similarly for the minimal primes.

Reply:
good point -- that was old Singular style (never give back an empty list) instead of Oscar-style.